mathics/main.py

```diff
@@ -451,7 +462,15 @@ def dump_tracing_stats():
                 print(query)
             result = evaluation.evaluate(query, timeout=settings.TIMEOUT)
             if result is not None:
-                shell.print_result(result, strict_wl_output=args.strict_wl_output)
+                head = result.last_eval.get_head()
+                form = ""
+                if head in evaluation.definitions.printforms:
```

Comment:
It appears that there are other forms not in `$PrintForms` that till get rendered `//XXXForm=`. In particular `QuantityForm` or `ScientificForm`.  But others like `StringForm` don't work that way. 

Although this PR came first, it  should come after and build on #567 .  With that the separation of Form classes like those in `$PrintForms` versus those like `StringForm` can be marked inside the code, say as a class variable. And then the BaseForm class could gather those marked into something like `printforms` but is more appropriate to front-end marking `//XXXForm`. 

Is there something that distinguishes `StringForm` kinds of forms from those that appear in as `//`?  Is there something that distinguishes ``$PrintForm` items from those things that can appear after `//`?

Reply:
You are right. It should be for items of `$OutputForms` instead of `$PrintForms`.
Both should be updated when a new format rule is set.

Reply:
About `StringForm`, maybe it is because there is not a clear "underlying" expression associated to the form, like in the other Forms. For example, `BaseForm[10,2]` means `10`, and it is shown in binary representation. In `StringForm["``+``", a,b]`,
`"``+``"` does not mean anything without the other elements.